### DISCUSSION TITLE
Github doesn't display new travis builds near the commits

### DISCUSSION THREAD

**Discussion by lutaev** [community/community#23302] [New to GitHub]
<p>Hi there.</p>
<p>I’m working on a simple app and got a problem with travis builds. I created a branch and <a href="https://github.com/lutaev/richi/pull/3" rel="nofollow noopener">a new pull-request</a> with one commit and my build was failed as expected. When I fixed issues, I tried to push more commits and found that travis doesn’t react on it. Then I push more commits with the same results - no travis reaction. I checked my app on <a href="http://travis-ci.org" rel="nofollow noopener">travis-ci.org</a> and found that travis successfully created build on each commit. But github shows me only first failed build (see image below). Here is my .travis.yml file:</p>
<pre><code>sudo: required
language: node_js
node_js:
- '13'
env:
  global:
  - PG_SQL_USER=hsxyxlxq
  - PG_SQL_HOST=balarama.db.elephantsql.com
  - PG_SQL_DB=hsxyxlxq
  - PG_SQL_TEST_USER=gskibphv
  - PG_SQL_TEST_HOST=drona.db.elephantsql.com
  - PG_SQL_TEST_DB=gskibphv
  - secure: #####
  - secure: #####
before_script:
- npm install
script:
- npm run migrate:test
- npm run test
deploy:
  provider: heroku
  app: richi-sh
  api_key:
    secure: #####
  on:
    repo: lutaev/richi
</code></pre>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/1X/cd30a4dbb6b40a78dd40fe7330d2af18ca171a47.png" data-download-href="/uploads/short-url/thc6f1EMfGAMKZHIqF7ETL0mtzF.png?dl=1" title="Screenshot 2020-04-13 at 20.02.38.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/6015401/181076082-9fc7b267-8bdc-46ed-a317-dbb444585d75.png" alt="Screenshot 2020-04-13 at 20.02.38.png" data-base62-sha1="thc6f1EMfGAMKZHIqF7ETL0mtzF" width="690" height="317" srcset="https://aws1.discourse-cdn.com/github/optimized/1X/cd30a4dbb6b40a78dd40fe7330d2af18ca171a47_2_690x317.png, https://aws1.discourse-cdn.com/github/original/1X/cd30a4dbb6b40a78dd40fe7330d2af18ca171a47.png 1.5x, https://aws1.discourse-cdn.com/github/original/1X/cd30a4dbb6b40a78dd40fe7330d2af18ca171a47.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/1X/cd30a4dbb6b40a78dd40fe7330d2af18ca171a47_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screenshot 2020-04-13 at 20.02.38.png</span><span class="informations">763×351 23.7 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>I’m new in CI, what should I do in this case?</p>
<p>Thanks.</p>

**Comment by francisfuzz** [comment, marked as answer]
<p><a class="mention" href="/lutaev">@lutaev</a> - The next best step is to reach out to Travis CI about why those particular commit statuses weren’t created for each of those commits’ builds.</p>
<p>Here’s some additional context that supports that step. GitHub provides the interface for third-party integrators (like Travis CI) to report the statuses of their builds via our GitHub’s REST API’s <a href="https://developer.github.com/v3/repos/statuses" rel="nofollow noopener">Statuses API</a>.</p>
<p>However, GitHub doesn’t create those statuses — the respective third-party integrator creates those statuses using this endpoint:</p>
<p><a href="https://developer.github.com/v3/repos/statuses/#create-a-status" rel="nofollow noopener">https://developer.github.com/v3/repos/statuses/#create-a-status</a></p>
<p>This pull request, <a href="https://github.com/lutaev/richi/pull/3" rel="nofollow noopener">https://github.com/lutaev/richi/pull/3</a>, has a head branch proposing changes from these four commits:</p>
<ul>
<li><a href="https://github.com/lutaev/richi/pull/3/commits/6da080254780cbdd541e96a9a741b5d9ad74db83" rel="nofollow noopener"><code>6da0802</code></a></li>
<li><a href="https://github.com/lutaev/richi/pull/3/commits/80bf9b26839d84908b838417025a95586545ddb8" rel="nofollow noopener"><code>80bf9b2</code></a></li>
<li>
<a href="https://github.com/lutaev/richi/pull/3/commits/db631a1639542f726af871e61e6b9a9ff3060359" rel="nofollow noopener"><code>db631a1</code></a> </li>
<li><a href="https://github.com/lutaev/richi/pull/3/commits/b6d1eb5dccaf6e3818404aa45888e88391fa3013" rel="nofollow noopener"><code>b6d1eb5</code></a></li>
</ul>
<p>At this time of writing, only<a href="https://github.com/lutaev/richi/pull/3/commits/6da080254780cbdd541e96a9a741b5d9ad74db83" rel="nofollow noopener"><code>6da0802</code></a> has commit statuses created for it. There are two ways for inspecting its statuses.</p>
<p>The first way is by navigating to the commit independent of the pull request context will highlight those statuses combined “state” next to the commit message:</p>
<p><a href="https://github.com/lutaev/richi/commit/6da080254780cbdd541e96a9a741b5d9ad74db83" rel="nofollow noopener">https://github.com/lutaev/richi/commit/6da080254780cbdd541e96a9a741b5d9ad74db83</a></p>
<p>The second way is to use the Statuses API, specifically either of these endpoints:</p>
<ul>
<li><a href="https://developer.github.com/v3/repos/statuses/#list-statuses-for-a-specific-ref" rel="nofollow noopener">https://developer.github.com/v3/repos/statuses/#list-statuses-for-a-specific-ref</a></li>
<li><a href="https://developer.github.com/v3/repos/statuses/#get-the-combined-status-for-a-specific-ref" rel="nofollow noopener">https://developer.github.com/v3/repos/statuses/#get-the-combined-status-for-a-specific-ref</a></li>
</ul>
<p>Here’s an example listing all the statuses for that commit:</p>
<ul>
<li><a href="https://api.github.com/repos/lutaev/richi/commits/6da080254780cbdd541e96a9a741b5d9ad74db83/statuses" rel="nofollow noopener">https://api.github.com/repos/lutaev/richi/commits/6da080254780cbdd541e96a9a741b5d9ad74db83/statuses</a></li>
</ul>
<p>The same approach can be applied for fetching the combined status for that ref.</p>
<p>Whichever approach you use, these statuses are appropriately attributed to Travis CI in the context of the respective builds. </p>
<p>I hope this information helps!</p>

**Comment by lutaev** [comment]
<p>Thank you very much. I found the same topic on <a href="http://travic-ci.org" rel="nofollow noopener">travic-ci.org</a> help community. I followed the instructions from this topic and get success.</p>
<p>Link to topic <a href="https://travis-ci.community/t/travis-ui-does-not-link-a-build-to-correct-commit-on-github-pr/7711" rel="nofollow noopener">https://travis-ci.community/t/travis-ui-does-not-link-a-build-to-correct-commit-on-github-pr/7711</a></p>
